src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -711,9 +711,35 @@ private void closeTempFile(String encodedPartition) {
   }
 
   private void closeTempFile() {
+    ConnectException connectException = null;
     for (String encodedPartition : tempFiles.keySet()) {
       // Close the file and propagate any errors
-      closeTempFile(encodedPartition);
+      try {
+        closeTempFile(encodedPartition);
+      } catch (ConnectException e) {
+        // still want to close all of the other data writers
+        connectException = e;
+        log.error(
+            "Failed to close temporary file for partition " + encodedPartition
+                + ". The connector will attempt to rewrite the temporary file."
+        );
+      }
+    }
+
+    if (connectException != null) {
+      // at least one tmp file did not close properly
+      for (String encodedPartition : tempFiles.keySet()) {
+        deleteTempFile(encodedPartition);
```

Comment:
what if delete fails with `ConnectException` too? Are we still going to have the same problem? Should we try to delete all the files at least?

Reply:
I have added more graceful delete however if it fails on a delete that shouldn't result in any empty files because the file will not be appended to the WAL so it will never be committed. Worst case that there will be some stale `tmp` files